### DISCUSSION TITLE
Example of an Advanced Setup in Obsidian

### DISCUSSION THREAD

**Discussion by qt-dork** [out-of-cheese-error/gooseberry#73] [Show and tell]
idk @Ninjani told me this when i sent her some pics of my gooseberry setup and templating, so i figured i should do what she asked and post this on the discussion page, like she wanted.

> Whoa looks really great!
> Ooh, would you feel up to posting something like this on the github discussions page to give people an idea of an advanced setup?

i had a lot of fun messing around and learning a bunch of handlebars stuff to make this so I hope it's useful for y'all too? it's set up to include the *(Show in context)* link inline next to whatever would be most relevant, because I feel like it's not useful to see a link to view the context at the very bottom away from everything else. it also doesn't include tags in annotations that doesn't have any, which i find useful because i really only use annotation tags in page notes that apply tags to the actual page in Obsidian. really, the only thing left i'd like to add is links to the related daily note for that annotation or page, but chronos for rust isn't as comprehensive as moment.js, and it doesn't support roam-styled dates (e.g. *April 6th, 2021*), so i guess i'll fix that if they're ever supported. 

### Screenshots

![](https://cdn.discordapp.com/attachments/825905439949586432/828760979562299412/Screen_Shot_2021-04-05_at_3.39.50_PM.png)
Screenshot of a part of a page.

![](https://cdn.discordapp.com/attachments/825905439949586432/828761288691023892/Screen_Shot_2021-04-05_at_3.41.22_PM.png)
Example of the formatting in a highlight with an annotation added.

![](https://cdn.discordapp.com/attachments/825905439949586432/828761526184968242/Screen_Shot_2021-04-05_at_3.42.16_PM.png)
Example of the formatting in a page note.

#### Individual annotation template:
```
annotation_template = '''

### {{id}}
{{~#if tags}}
Tags: {{#each tags}}#{{this}}{{#unless @last}}, {{/unless}}{{/each}}
{{~/if}}

{{#each highlight}}> {{this}}{{/each}} {{#if highlight}}([See in context]({{incontext}})){{/if}}

{{text}}{{#unless highlight}} ([See in context]({{incontext}})){{/unless}}
'''
```

#### Page template:
```
page_template = '''
# {{raw_annotations.0.title}}

## Highlights
{{#each annotations}}{{this}}{{/each}}
## Source

[{{raw_annotations.0.title}}]({{raw_annotations.0.uri}})
'''
```

there's some minor issues with spacing, as there'll sometimes be too many or too few line breaks in stuff, but i don't really know how to use the tilde in handlebars for line breaks, so if anyone can help fix that i'd really appreciate it

**Comment by mberlinger3** [comment]
Haha these look like my class notes! Loving this template.
